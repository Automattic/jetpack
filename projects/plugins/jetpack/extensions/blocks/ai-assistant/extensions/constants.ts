// All Jetpack Form blocks to extend
export const JETPACK_FORM_CHILDREN_BLOCKS = [
	'jetpack/field-name',
	'jetpack/field-email',
	'jetpack/field-text',
	'jetpack/field-textarea',
	'jetpack/field-checkbox',
	'jetpack/field-date',
	'jetpack/field-telephone',
	'jetpack/field-url',
	'jetpack/field-checkbox-multiple',
	'jetpack/field-radio',
	'jetpack/field-select',
	'jetpack/field-consent',
	'jetpack/button',
] as const;

export const EXTENDED_BLOCKS = [
	'core/heading',
	'core/paragraph',
	'core/list-item',
	'core/list',
	'jetpack/contact-form',
	...JETPACK_FORM_CHILDREN_BLOCKS,
];

export type ExtendedBlockProp =
	| 'core/heading'
	| 'core/paragraph'
	| 'core/list-item'
	| 'core/list'
	| 'jetpack/contact-form'
	| ( typeof JETPACK_FORM_CHILDREN_BLOCKS )[ number ];
