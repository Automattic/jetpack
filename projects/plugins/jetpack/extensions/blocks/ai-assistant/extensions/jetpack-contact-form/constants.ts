export const JETPACK_FORM_AI_COMPOSITION_EXTENSION = 'ai-assistant-form-support' as const;

export const isJetpackFromBlockAiCompositionAvailable =
	window?.Jetpack_Editor_Initial_State?.available_blocks?.[ JETPACK_FORM_AI_COMPOSITION_EXTENSION ]
		?.available;

// All blocks to extend
export const JETPACK_FORM_CHILDREN_BLOCKS = [
	'jetpack/field-name',
	'jetpack/field-email',
	'jetpack/field-text',
	'jetpack/field-textarea',
	'jetpack/field-checkbox',
	'jetpack/field-date',
	'jetpack/field-telephone',
	'jetpack/field-url',
	'jetpack/field-checkbox-multiple',
	'jetpack/field-radio',
	'jetpack/field-select',
	'jetpack/field-consent',
	'jetpack/button',
];
