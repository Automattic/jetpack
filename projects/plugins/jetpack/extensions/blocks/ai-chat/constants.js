import { __ } from '@wordpress/i18n';

export const DEFAULT_ASK_BUTTON_LABEL = __( 'Ask', 'jetpack' );
export const DEFAULT_PLACEHOLDER = __( 'Ask a question about this site', 'jetpack' );
