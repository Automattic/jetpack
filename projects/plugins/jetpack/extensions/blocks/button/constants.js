/* eslint-disable wpcalypso/import-docblock */
/**
 * WordPress dependencies
 */
import { __experimentalUseGradient as useGradient } from '@wordpress/block-editor';

export const IS_GRADIENT_AVAILABLE = !! useGradient;

export const INITIAL_BORDER_RADIUS_POSITION = 5;
export const MAX_BORDER_RADIUS_VALUE = 50;
export const MIN_BORDER_RADIUS_VALUE = 0;
