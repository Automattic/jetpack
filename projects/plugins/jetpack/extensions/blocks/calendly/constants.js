export const CALENDLY_EXAMPLE_URL = 'https://calendly.com/wpcom/jetpack-block-example';
