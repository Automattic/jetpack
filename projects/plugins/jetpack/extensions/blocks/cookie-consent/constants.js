import { __, sprintf } from '@wordpress/i18n';

export const COOKIE_POLICY_URL = 'https://automattic.com/cookies/';

export const DEFAULT_TEXT = sprintf(
	// translators: %s is a link to the cookie policy.
	__(
		'Privacy & Cookies: This site uses cookies. By continuing to use this website, you agree to their use. To find out more, including how to control cookies, see here: %s.',
		'jetpack'
	),
	`<a href="${ COOKIE_POLICY_URL }">${ __( 'Cookie Policy', 'jetpack' ) }</a>`
);
