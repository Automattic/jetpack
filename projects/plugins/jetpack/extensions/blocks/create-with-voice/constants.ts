export const JETPACK_CREATE_WITH_VOICE_EXTENSION = 'create-with-voice' as const;
