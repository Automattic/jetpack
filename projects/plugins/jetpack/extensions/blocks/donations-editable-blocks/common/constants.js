export const ONE_TIME_DONATION_TAB = 'oneTimeDonation';
export const MONTHLY_DONATION_TAB = 'monthlyDonation';
export const ANNUAL_DONATION_TAB = 'annualDonation';
export const DEFAULT_TAB = ONE_TIME_DONATION_TAB;
