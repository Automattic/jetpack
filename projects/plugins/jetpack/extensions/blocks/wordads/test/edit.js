import { useModuleStatus } from '@automattic/jetpack-shared-extension-utils';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { AD_FORMATS, DEFAULT_FORMAT } from '../constants';
import WordAdsEdit from '../edit';
import wideSkyscraperExample from '../example_160x600.png';
import rectangleExample from '../example_300x250.png';
import mobileLeaderboardExample from '../example_320x50.png';
import leaderboardExample from '../example_728x90.png';

jest.mock( '@automattic/jetpack-shared-extension-utils' );

describe( 'WordAdsEdit', () => {
	const moduleStatus = {
		isModuleActive: true,
		changeStatus: jest.fn(),
	};
	beforeEach( () => {
		useModuleStatus.mockReturnValue( { ...moduleStatus } );
	} );
	afterEach( () => {
		jest.clearAllMocks();
	} );

	const defaultAttributes = { format: DEFAULT_FORMAT };
	const defaultProps = { attributes: defaultAttributes };

	const getFormat = format => AD_FORMATS.find( ( { tag } ) => tag === format );

	const renderWordAdsEdit = props => {
		const { container } = render( <WordAdsEdit { ...props } /> );

		// eslint-disable-next-line testing-library/no-node-access,testing-library/no-container
		return container.querySelector( '.jetpack-wordads__ad' );
	};

	// Renders the component, extracting the inner placeholder element and finding
	// selected format object styles are created from.
	const renderFormatted = format => {
		// eslint-disable-next-line testing-library/render-result-naming-convention -- False positive.
		const placeholder = renderWordAdsEdit( {
			...defaultProps,
			attributes: { format },
		} );
		const selectedFormat = getFormat( format );

		return { placeholder, selectedFormat };
	};

	test( 'matches snapshot', () => {
		const { container } = render( <WordAdsEdit { ...defaultProps } /> );

		expect( container ).toMatchSnapshot( 'WordAdsEdit' );
	} );

	test( 'renders ad placeholder with correct css class and styles', () => {
		// eslint-disable-next-line testing-library/render-result-naming-convention -- False positive.
		const placeholder = renderWordAdsEdit( defaultProps );
		const selectedFormat = getFormat( defaultAttributes.format );

		expect( placeholder ).toHaveStyle( `width: ${ selectedFormat.width }px` );
		expect( placeholder ).toHaveStyle( `height: ${ selectedFormat.height }px` );
		expect( placeholder ).toHaveStyle( `backgroundImage: url( ${ rectangleExample } )` );
		expect( placeholder ).toHaveStyle( `backgroundSize: cover` );
	} );

	test( 'renders leaderboard format correctly', () => {
		const { placeholder, selectedFormat } = renderFormatted( 'leaderboard' );

		expect( placeholder ).toHaveStyle( `width: ${ selectedFormat.width }px` );
		expect( placeholder ).toHaveStyle( `height: ${ selectedFormat.height }px` );
		expect( placeholder ).toHaveStyle( `backgroundImage: url( ${ leaderboardExample } )` );
	} );

	test( 'renders mobile_leaderboard format correctly', () => {
		const { placeholder, selectedFormat } = renderFormatted( 'mobile_leaderboard' );

		expect( placeholder ).toHaveStyle( `width: ${ selectedFormat.width }px` );
		expect( placeholder ).toHaveStyle( `height: ${ selectedFormat.height }px` );
		expect( placeholder ).toHaveStyle( `backgroundImage: url( ${ mobileLeaderboardExample } )` );
	} );

	test( 'renders wideskyscraper format correctly', () => {
		const { placeholder, selectedFormat } = renderFormatted( 'wideskyscraper' );

		expect( placeholder ).toHaveStyle( `width: ${ selectedFormat.width }px` );
		expect( placeholder ).toHaveStyle( `height: ${ selectedFormat.height }px` );
		expect( placeholder ).toHaveStyle( `backgroundImage: url( ${ wideSkyscraperExample } )` );
	} );

	test( 'renders placeholder and activates the plugin', async () => {
		useModuleStatus.mockReturnValue( { ...moduleStatus, isModuleActive: false } );
		render( <WordAdsEdit { ...defaultProps } /> );

		const activateButton = screen.getByText( 'Activate WordAds' );
		await userEvent.click( activateButton );
		expect( moduleStatus.changeStatus ).toHaveBeenCalled();
	} );
} );
