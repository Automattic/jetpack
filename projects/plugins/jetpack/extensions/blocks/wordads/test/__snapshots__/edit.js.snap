// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordAdsEdit matches snapshot: WordAdsEdit 1`] = `
<div>
  <div
    aria-label="Block: undefined"
    class="block-editor-block-list__block wp-block"
    id="block-undefined"
    role="document"
    tabindex="0"
  >
    <div
      class="wp-block-jetpack-wordads jetpack-wordads-mrec"
    >
      <div
        class="jetpack-wordads__ad"
        style="width: 300px; height: 250px; background-size: cover;"
      />
    </div>
  </div>
</div>
`;
